### PR TITLE
[11.0] [ADD] cms_mrp_bom_form

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
Depends on [https://github.com/OCA/website/pull/481](url)

### CODE REVIEW

File: cms_mrp_bom_form/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'CMS Form MRP Boms',
+    'summary': """
+        Search forms for MRP Boms""",
+    'version': '11.0.1.0.0',
+    'license': 'AGPL-3',
```
Comment (MeetKD) [inline cms_mrp_bom_form/__manifest__.py:10]: Can you please add website key ?

File: cms_mrp_bom_form/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'CMS Form MRP Boms',
+    'summary': """
+        Search forms for MRP Boms""",
```
Comment (MeetKD) [inline cms_mrp_bom_form/__manifest__.py]: search forms for MRP Boms in website !

**Review by MeetKD — CHANGES_REQUESTED** [review]

File: cms_mrp_bom_form/README.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+This module allows website visitors to search and browse through the list of
+Bills of Materials that the company has decided to publish.
+
+
+Usage
+=====
+
+Go on the website app of Odoo. Then you can reach the following pages:
+
+* /mrp_boms: shows you a nice form to search and list Bills of Materials.
```
Comment (simahawk) [inline cms_mrp_bom_form/README.rst:18]: Do you foresee other records from mrp area that you might want to publish? 
If yes, then I'd prefix all the routes, like: `/mrp/boms`.

File: cms_mrp_bom_form/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+from odoo import http
+from odoo.addons.cms_form.controllers.main import SearchFormControllerMixin
+
+
+class MrpBomFormController(http.Controller, SearchFormControllerMixin):
+    """Product form controller."""
+
+    @http.route([
+        '/mrp_boms',
```
Comment (simahawk) [inline cms_mrp_bom_form/controllers/main.py]: you should set this route as `cms_search_url` on mrp.bom

File: cms_mrp_bom_form/models/cms_form.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+from odoo import fields, models
+
+
+class MrpBomFormSearch(models.AbstractModel):
+
+    _name = 'cms.form.search.mrp.bom'
+    _inherit = 'cms.form.search'
+    _form_model = 'mrp.bom'
+    _form_model_fields = ('name', 'code', )
+    _form_required_fields = ('name', 'code', )
```
Comment (simahawk) [inline cms_mrp_bom_form/models/cms_form.py]: do you really want search fields to be required?

File: cms_mrp_bom_form/models/cms_form.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    name = fields.Char()
```
Comment (simahawk) [inline cms_mrp_bom_form/models/cms_form.py]: you don't need this, unless you are overriding some field attributes :)

File: cms_mrp_bom_form/models/cms_form.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+    def form_search_domain(self, search_values):
+        domain = super().form_search_domain(search_values)
+        is_website_publisher = self.env['res.users'].has_group(
+            'website.group_website_publisher')
+        if not is_website_publisher:
+            default_domain = [
+                ('website_published', '=', True),
```
Comment (simahawk) [inline cms_mrp_bom_form/models/cms_form.py]: wouldn't be better to rely on ACL + record rule?

**Review by simahawk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**